### PR TITLE
Add examples for LinkedList

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of #29348.

r? @steveklabnik

### CODE REVIEW

**Comment by frewsxcv** [comment]
```
failures:
---- linked_list::LinkedList<T>::iter_mut_0 stdout ----
	thread 'linked_list::LinkedList<T>::iter_mut_0' panicked at 'test executable failed:
thread 'main' panicked at 'assertion failed: `(left == right)` (left: `Some(10)`, right: `Some(0)`)', <anon>:17
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

File: src/libcollections/linked_list.rs [hunk 1]
```diff
@@ -238,6 +264,28 @@ impl<T> LinkedList<T> {
...
+    /// list.push_back(2);
+    ///
+    /// for element in list.iter_mut() {
+    ///     *element += 10;
+    /// }
+    ///
+    /// let mut iter = list.iter();
+    /// assert_eq!(iter.next(), Some(&0));
+    /// assert_eq!(iter.next(), Some(&1));
+    /// assert_eq!(iter.next(), Some(&2));
```
Comment (frewsxcv) [inline src/libcollections/linked_list.rs]: Values needed to be updated here to account for `+= 10`.

Comment (GuillaumeGomez) [inline src/libcollections/linked_list.rs]: Oh damn, thanks!

Comment (GuillaumeGomez) [inline src/libcollections/linked_list.rs]: This note comment still appears because it wasn't on the incorrect line. :p

**Comment by alexcrichton** [comment]
@bors: retry

* #34595

**Pull request merged**